bee-bundle/src/transaction/transaction.rs

```diff
@@ -336,6 +336,45 @@ impl TransactionBuilder {
         Self::default()
     }
 
+    pub fn essence(&self) -> TritBuf {
+        let mut essence = TritBuf::<T1B1Buf>::new();
+        let address = self.address.clone().unwrap();
+        let value = TritBuf::from(*self.value.clone().unwrap().to_inner());
+        let tag = self.tag.clone().unwrap();
+        let timestamp = TritBuf::from(*self.timestamp.clone().unwrap().to_inner() as i64);
+        let index = TritBuf::from(*self.index.clone().unwrap().to_inner() as i64);
+        let last_index = TritBuf::from(*self.last_index.clone().unwrap().to_inner() as i64);
+
+        address.to_inner().iter().for_each(|x| essence.push(*x));
```

Comment:
We should definitely avoid all these `push`, it makes it very inefficient.
Try to use a pre-allocated array full of 0 and copy maybe ?